addons/html_editor/controllers/main.py

```diff
@@ -591,8 +591,11 @@ def link_preview_metadata_internal(self, preview_url):
             Actions = request.env['ir.actions.actions']
             context = dict(request.env.context)
             words = preview_url.strip('/').split('/')
-
-            record_id = int(words.pop())
+            # If there is no recordID in url return empty dict.
+            try:
+                record_id = int(words.pop())
+            except ValueError:
+                return {}
```

Comment:
It s a new issue ? Or you don t need this fix ? (Maybe do it in stable version)

Reply:
Actually current method for fetching internal metadata has been implemented such a way that, it assumes we will have URL something like "/project/task/1" or something like that, what if the method is called with URL like "/test/something" in that case it throws ValueError.

Either we should not call ```/html_editor/link_preview_internal``` at all by adding few check in javascript or we should handle ValueError.

Reply:
@rahg-odoo Please adapt test case like we did here: https://github.com/odoo-dev/odoo/pull/4620/files